### ISSUE TITLE
caja gksu: apps launched by caja gksu are inaccessible to orca

### ISSUE THREAD

**Issue by coffeeking** [mate-desktop/caja-extensions#21]
steps to reproduce: Enable orca.  Add an application to the desktop by highlighting it in the menus, right clicking it and selecting add this launcher to the desktop. Right click on the new icon and select "run as administrator". Expected results: orca should read the new application. Orca is silent, and can't see the window. In fact, it won't read anything until the application is closed, at which point it resumes as if you'd never launched it.

**Comment by monsta** [comment]
That's the question to Orca devs. Can it read anything from app that's run as root?

**Comment by coffeeking** [comment]
hi

The problem is universal. Orca can't read any apps run as root because 
it can't get at them to read them. It's a problem we've been struggling 
to figure out. Policy kit helps a little bit but still, the only desktop 
that consistently gets around this is gnome shell, and I have no idea 
what they do to achieve it.

Thanks

Kendell Clark

**Comment by coffeeking** [comment]
hi
The solution seems to lie in gnome settings-daemon. They have code in
there to set up the gtk and qt modules for accessibility when an app is
launched, no matter what user starts the program. Would you guys mind
looking at gnome settings-daemon's code to see if you can figure out
what code needs to be added to mate settings-daemon to fix this? This is
not limited to running apps as root through caja-gksu, but any app run
as root in any way while mate is used, both the gtk2 and gtk3 versions.
The only reason I'm asking is because the manjaro guys have removed the
old installer sonar was using and now we have to use a qt based
installer, which is accessible but only when run as a non root user
because of this bug, so it's pretty important this bug get fixed. I'll
help in any way I can, I don't expect this to be done for me. If you
like I can try looking through the code myself and report what I find to
hear. Just let me know.
Thanks
Kendell Clark


kendell clark wrote:

**Comment by monsta** [comment]
> They have code in there to set up the gtk and qt modules for accessibility when an app is launched, no matter what user starts the program. 

I wonder how they did that - settings daemon is running as user...

**Comment by monsta** [comment]
Just to make sure, can you also check gnome-flashback? I wonder if it's really g-s-d (which is also used in flashback session) or maybe gnome-session (which isn't used there as flashback is a separate session on its own).

**Comment by coffeeking** [comment]
hi
I'd be glad to test flashback. Is it still maintained? I seem to
remember it being unmaintained since, I think it was gnome 3.6 but I'll
test that right now since I've got nothing else pressing to do.
Thanks
Kendell Clark


monsta wrote:

**Comment by monsta** [comment]
Yes, it's maintained now, there are some developers working on it. Stable release is at 3.20.x now.
